### PR TITLE
chore: rewrite changelog script, generate release changelog

### PR DESCRIPTION
**Pull request by Tigge**
Rewrite the changelog script in python and add support for additional features, such as generating a changelog for a release. This is now also used in the release workflow to automatically upload a release changelog to github.

### CODE REVIEW

File: sbin/release.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    utils.shell(f'yarn workspaces foreach version --deferred "{next_version}"')
+    utils.shell(f"yarn version apply --all")
+
+    print(" - Update changelog")
+    utils.shell(f'./sbin/changelog.py full --release "v{next_version}" > CHANGELOG.md')
+
+    print(" - Create release commit and tag")
+    utils.shell(f"git add -u")
+    utils.shell(f'git commit -m "v{next_version}"')
+    utils.shell(f'git tag -a "v{next_version}" -m "v{next_version}"')
```
Comment (steabert) [inline sbin/release.py:44]: This really needed to be python too? I would argue that at this point, there is no benefit :stuck_out_tongue:

File: sbin/changelog.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+import argparse
+import re
+import subprocess
+import sys
+
+import utils
```
Comment (steabert) [inline sbin/changelog.py:8]: Where does this module come from? I couldn't find it. Is it portable?

File: sbin/utils.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        "type": type,
+        "scope": scope,
+        "breaking": breaking == "!",
+        "description": header,
+    }
+
+
+def shell(cmd: str) -> str:
+    """ Call shell command and return the result """
+    return subprocess.check_output(cmd, shell=True, encoding="utf-8").strip()
```
Comment (steabert) [inline sbin/utils.py:29]: If everything is replaced with Python (cfr release script as well), then why not go the extra mile of replacing dependency on a shell?

File: sbin/release.py (hunk 1 above)
Comment (Tigge) [reply sbin/release.py]: Yeah agreed, I've removed `shell=True` now at least since that was quite silly.

File: sbin/utils.py (hunk 3 above)
Comment (Tigge) [reply sbin/utils.py]: Done

**Review by steabert — APPROVED** [review]

**Pull request merged**